### ISSUE TITLE
`PDS` java.io.IOException: Store of sourcecode.zip to s3 failed

### ISSUE THREAD

**Issue by Jeeppler** [mercedes-benz/sechub#1501]
# Problem

Uploading files to a local object storage does not work. I tested it using the `GoSec+PDS`-Solution and a local object storage (SeaweedFS). The exceptions are:

~~~
 com.mercedesbenz.sechub.commons.model.SecHubRuntimeException: Was not able to upload job data because of IO problems.
	at com.mercedesbenz.sechub.pds.job.PDSFileUploadJobService.handleUploadAndProblems(PDSFileUploadJobService.java:116)
	at com.mercedesbenz.sechub.pds.job.PDSFileUploadJobService.upload(PDSFileUploadJobService.java:84)
	at com.mercedesbenz.sechub.pds.job.PDSFileUploadJobService$$FastClassBySpringCGLIB$$6e4418a6.invoke(<generated>)
…
Caused by: java.io.IOException: Store of sourcecode.zip to s3 failed
	at com.mercedesbenz.sechub.storage.s3.aws.AwsS3JobStorage.store(AwsS3JobStorage.java:66)
	at com.mercedesbenz.sechub.pds.job.PDSFileUploadJobService.startUpload(PDSFileUploadJobService.java:184)
	at com.mercedesbenz.sechub.pds.job.PDSFileUploadJobService.handleUploadAndProblems(PDSFileUploadJobService.java:95)
…
Caused by: com.amazonaws.ResetException: Failed to reset the input stream;  If the request involves an input stream, the maximum stream buffer size can be configured via request.getRequestClientOptions().setReadLimit(int)
…
Caused by: java.io.IOException: Resetting to invalid mark
	at java.base/java.io.BufferedInputStream.reset(BufferedInputStream.java:454)
	at com.amazonaws.internal.SdkBufferedInputStream.reset(SdkBufferedInputStream.java:106)
	at com.amazonaws.internal.SdkFilterInputStream.reset(SdkFilterInputStream.java:120)
	at com.amazonaws.event.ProgressInputStream.reset(ProgressInputStream.java:168)
~~~

Uploading files to PDS with PDS versions 0.28.0 and less works well. Uploading files to PDS with PDS versions 0.29.0, 0.30.0 and 0.31.0 does not work.

This might be related to #1471.

NOTE: This bug only affects source uploads. Binary uploads work well.

**Comment by zigfridus** [comment]
@Jeeppler 
Please assign me to this issue. I'll examine it on my return.

**Comment by Jeeppler** [comment]
@zigfridus I assigned you.

**Comment by zigfridus** [comment]
Hello @Jeeppler 
As I understand you are working on the #1471 that could fix this issue #1501 now.  Thus I need your advice what should I do. Do I have to wait till you complete your work on #1471?
Thanks

**Comment by Jeeppler** [comment]
@zigfridus I am not sure, as this is a bug #1471 might solve it or might not solve it. My advice for you would be to work on a different issue, as the issue #1471 is currently my highest priority. As part of this issue a number of other things need to be improved as well.

My recommendation for you would be to work on #1506. This issue is an important step towards creating a web UI (the epic is: #11). We generate the OpenAPI client for Java, but it is not used until now. We have no idea if the Java client library is easy to use or does work well, therefore it would be nice to have a test/demo app.

**Comment by Jeeppler** [comment]
We fixed it with #1471.
